framework/auth/core.py

```diff
@@ -1203,12 +1203,27 @@ def merge_user(self, user):
             # Skip dashboard node
             if node.is_dashboard:
                 continue
-            node.add_contributor(
-                contributor=self,
-                permissions=node.get_permissions(user),
-                visible=node.get_visible(user),
-                log=False,
-            )
+            # if both accounts are contributor of the same project
+            if node.is_contributor(self) and node.is_contributor(user):
```

Comment:
It would be good to document how this conflict resolution works in the docstring of `merge_user`.